server/interpreter.js

```diff
@@ -2103,42 +2114,78 @@ Interpreter.prototype.throwException = function(value, opt_message) {
     }
     error = new this.Error(value, opt_message);
   }
-  this.executeException(error);
+  this.unwind(Interpreter.Completion.THROW, error, undefined);
   // Abort anything related to the current step.
   throw Interpreter.STEP_ERROR;
 };
 
 /**
- * Throw an exception in the interpreter that can be handled by an
- * interpreter try/catch statement.  If unhandled, a real exception will
- * be thrown.
- * @param {Interpreter.Value} error Value being thrown.
+ * Unwind the stack to the innermost relevant enclosing TryStatement,
+ * For/ForIn/WhileStatement or Call/NewExpression.  If this results in
+ * the stack being completely unwound the thread will be terminated
+ * and an appropriate error being logged.
+ * @param {Interpreter.Completion} type Completion type.
```

Comment:
I don't think so: Interpreter.Completion is an `@enum {number}`, so `null` should not be a valid argument and (just checking… Yes!) the compiler does correctly warn if you attempt to pass `null`.